mapproxy/source/ogcapimaps.py

```diff
@@ -0,0 +1,189 @@
+# This file is part of the MapProxy project.
+# Copyright (C) 2025 Spatialys
+#
+# Initial development funded by Centre National d'Etudes Spatiales (CNES): https://cnes.fr
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import sys
+from threading import Lock
+
+from mapproxy.client.http import HTTPClientError
+from mapproxy.request.base import BaseRequest
+from mapproxy.source import SourceError
+from mapproxy.source.wms import WMSLikeSource
+from mapproxy.srs import ogc_crs_url_to_auth_code, SRS, SupportedSRS
+from mapproxy.util.ogcapi import (
+    find_href_in_links,
+    normalize_srs_code,
+    build_absolute_url,
+)
+from mapproxy.util.py import reraise_exception
+
+import logging
+
+log = logging.getLogger("mapproxy.source.ogcapimaps")
+
+# For testing
+reset_cache = False
```

Comment:
I am not entirely happy with this global variable, but I can't think of a better way right now. But if you use it, please name it differently as the word `cache` is quite loaded in the MapProxy context.